frontend/src/metabase/setup/selectors.ts

```diff
@@ -118,13 +132,13 @@ export const getSteps = createSelector(
       tokenFeatures &&
       tokenFeatures["hosting"] &&
       !hasAddedPaidPlanInPreviousStep;
-    const shouldShowDataUsageStep = !isHosted;
+    const shouldShowDataUsageStep = !isEmbeddingUseCase && !isHosted;
 
     const steps: { key: SetupStep; isActiveStep: boolean }[] = [
-      { key: "welcome" as const },
-      { key: "language" as const },
+      !isEmbeddingUseCase && { key: "welcome" as const },
+      !isEmbeddingUseCase && { key: "language" as const },
       { key: "user_info" as const },
-      { key: "usage_question" as const },
+      !isEmbeddingUseCase && { key: "usage_question" as const },
       shouldShowDBConnectionStep && {
         key: "db_connection" as const,
       },
```

Comment:
all the conditionals are getting a bit messy here. what about doing something more like
```ts
if (isEmbeddingUseCase) {
	setup = [
	  { key: "user_info" as const },
	  { key: "completed" as const },
	];
}
```

Reply:
Definitely.
I'll clean this up.

Thanks